airflow/models/connection.py

```diff
@@ -43,6 +47,25 @@ def parse_netloc_to_hostname(*args, **kwargs):
     return _parse_netloc_to_hostname(*args, **kwargs)
 
 
+def sanitize_conn_id(conn_id: str | None) -> str | None:
+    """
+    Sanitises the connection id and allows only specific characters to be within. Namely,
+    it allows alphanumeric characters plus the symbols @,#,$,%,&,!,-,_, and () from 1
+    and up to 200 consecutive matches.
+
+    The character selection is such that it prevents the injection of javascript or
+    executable bits in order to avoid any awkward behavior in the front-end.
+
+    :param conn_id: The connection id to sanitize.
+    :return: the sanitized string, `None` otherwise.
+    """
+    res = None
+    if conn_id is None or (res := re.match(_RE_SANITIZE_CONN_ID, conn_id)) is None:
+        log.warning("We failed to match `conn_id` to the allowed pattern or it was None")
+
+    return res if res is None else conn_id
```

Comment:
```suggestion
    if conn_id is None or (res := re.match(_RE_SANITIZE_CONN_ID, conn_id)) is None:
        log.warning("We failed to match `conn_id` to the allowed pattern or it was None")
        return conn_id
    return res
```

Reply:
no strong opinions against that, so happy to commit that change - always in favour of reducing the LoC in a commit :)